### DISCUSSION TITLE
browser mode: component is not shown in browser

### DISCUSSION THREAD

**Discussion by nsulzberger** [vitest-dev/vitest#5370] [Q&A] [answered]
I am using vitest in browser mode and would like to see my rendered component in the browser.
I am adding my component to the test iframe with:
```
const container = document.createElement("div");
...
 document.body.appendChild(container);
```

When I look at the document of the browser I can see the following structure: 
```
<body>
    <iframe id="vitest-ui" src=""></iframe>
    <div id="vitest-tester">
	<iframe loading="eager" src="<path to test 1>">
	</iframe><iframe loading="eager" src="<path to test 2>"></iframe>
	<iframe loading="eager" src="<path to test 3>"></iframe>
	</div>
</body>
```

The iframe with id "vitest-ui" has a height of 100vh, so the iFrames with the test results are actually never displayed. When I set it to 0, I can see my rendered component in one of the iframes. 

I was watching this video tutorial, and there you can see the rendered content:
https://www.youtube.com/watch?v=d5S7Dy0rpQo
 Is this not supported anymore after the isolation with iframes has been added?

my vitest config:
```
{
  setupFiles: ["./tests/vitest/setup.ts"],
  include: ["./tests/vitest/**/*.spec.ts"],
  reporters: isCI ? [["junit", {
    suite: "vitest",
    useBrowserName: false
  }]] : ["default", "html"],
  outputFile: {
    junit: "./vitest-report.xml"
  },
  browser: {
    enabled: true,
    headless: false,
    name: "chromium",
    provider: "playwright",
    providerOptions: {
      launch: {
        args: ["--ignore-certificate-errors"]
      }
    },
    fileParallelism: false,
    isolate: false
  }
```

**Comment by userquin** [comment]
The UI part still not included

**Reply by nsulzberger** [reply]
Thanks a lot for your fast reply! I am sorry but I am very new to vitest. 
Are you referring to the comment here? https://github.com/vitest-dev/vitest/pull/3584#issuecomment-1949890011. Is the UI only a planned feature, and has never been implemented in a release? Or can I use an older release?

**Reply by userquin** [reply, marked as answer]
> Are you referring to the comment here?

Yes. 

> Is the UI only a planned feature, and has never been implemented in a release? Or can I use an older release?

To show the test UI we need to include the logic in Vitest UI, `browser` feature only includes the logic to run the test in the browser; rn there is no Vitest version with the test UI tab, only that PR includes it, and yes, in a future the tab should be there

**Reply by nsulzberger** [reply]
cool, great to hear that it will come at some point in the future!! What is missing that it cannot be merged yet?

**Reply by userquin** [reply]
The PR is too old, too many changes in the core.

**Reply by userquin** [reply]
@nsulzberger check https://github.com/vitest-dev/vitest/discussions/5828

**Reply by nsulzberger** [reply]
Thanks for the hint!
